prow/cmd/generic-autobumper/main.go

```diff
@@ -19,47 +19,467 @@ package main
+
+// PRTitleBody returns the body of the PR, this function runs after each commit
+func (c *client) PRTitleBody() (string, string, error) {
+	return makeCommitSummary(c.o.Prefixes, c.versions), "", nil
+}
+
+// Options is the options for autobumper operations.
+type Options struct {
+	// The URL where upstream image references are located. Only required if Target Version is "upstream" or "upstreamStaging". Use "https://raw.githubusercontent.com/{ORG}/{REPO}"
+	// Images will be bumped based off images located at the address using this URL and the refConfigFile or stagingRefConigFile for each Prefix.
+	UpstreamURLBase string `yaml:"upstreamURLBase"`
+	// The config paths to be included in this bump, in which only .yaml files will be considered. By default all files are included.
+	IncludedConfigPaths []string `yaml:"includedConfigPaths"`
+	// The config paths to be excluded in this bump, in which only .yaml files will be considered.
+	ExcludedConfigPaths []string `yaml:"excludedConfigPaths"`
+	// The extra non-yaml file to be considered in this bump.
+	ExtraFiles []string `yaml:"extraFiles"`
+	// The target version to bump images version to, which can be one of latest, upstream, upstream-staging and vYYYYMMDD-deadbeef.
+	TargetVersion string `yaml:"targetVersion"`
+	// List of prefixes that the autobumped is looking for, and other information needed to bump them. Must have at least 1 prefix.
+	Prefixes []Prefix `yaml:"prefixes"`
+}
+
+// Prefix is the information needed for each prefix being bumped.
+type Prefix struct {
+	// Name of the tool being bumped
+	Name string `yaml:"name"`
+	// The image prefix that the autobumper should look for
+	Prefix string `yaml:"prefix"`
+	// File that is looked at to determine current upstream image when bumping to upstream. Required only if targetVersion is "upstream"
+	RefConfigFile string `yaml:"refConfigFile"`
+	// File that is looked at to determine current upstream staging image when bumping to upstream staging. Required only if targetVersion is "upstream-staging"
+	StagingRefConfigFile string `yaml:"stagingRefConfigFile"`
+	// The repo where the image source resides for the images with this prefix. Used to create the links to see comparisons between images in the PR summary.
+	Repo string `yaml:"repo"`
+	// Whether or not the format of the PR summary for this prefix should be summarised.
+	Summarise bool `yaml:"summarise"`
+	// Whether the prefix tags should be consistent after the bump
+	ConsistentImages bool `yaml:"consistentImages"`
+}
+
+func parseOptions() (*Options, *bumper.Options, error) {
 	var config string
 	var labelsOverride []string
+	var skipPullRequest bool
 
 	flag.StringVar(&config, "config", "", "The path to the config file for the autobumber.")
 	flag.StringSliceVar(&labelsOverride, "labels-override", nil, "Override labels to be added to PR.")
+	flag.BoolVar(&skipPullRequest, "skip-pullrequest", false, "")
 	flag.Parse()
 
-	var o bumper.Options
+	var o Options
+	var pro bumper.Options
```

Comment:
@alvaroaleman , saw your comment in https://github.com/kubernetes/test-infra/pull/21952#discussion_r627712028, this is less easier to do in this PR due to duplicated fields for backwards compatibility(If deleting these field from bumper.Options then it'll break the old set of functions). I think it's doable and will do it in the next PR for cleaning up